localstack/services/cloudformation/models/apigateway.py

```diff
@@ -166,12 +167,45 @@ def _create(resource_id, resources, resource_type, func, stack_name):
             result = client.create_rest_api(**kwargs)
 
             body = props.get("Body")
-            if body:
+            s3_body_location = props.get("BodyS3Location", {})
+            if body or s3_body_location:
                 # the default behavior for imports via CFn is basepath=ignore (validated against AWS)
-                api_params = {"basepath": "ignore"}
-                body = json.dumps(body) if isinstance(body, dict) else body
+                import_parameters = props.get("Parameters", {})
+                import_parameters.setdefault("basepath", "ignore")
+
+                if body:
+                    body = json.dumps(body) if isinstance(body, dict) else body
+                else:
+                    get_obj_kwargs = {}
+                    if version_id := s3_body_location.get("Version"):
+                        get_obj_kwargs["VersionId"] = version_id
+
+                    # what is the approach when client call fail? Do we bubble it up?
+                    s3_client = aws_stack.connect_to_service("s3")
+                    get_obj_req = s3_client.get_object(
+                        Bucket=s3_body_location.get("Bucket"),
+                        Key=s3_body_location.get("Key"),
+                        **get_obj_kwargs,
+                    )
+                    if etag := s3_body_location.get("ETag"):
+                        if etag != get_obj_req["ETag"]:
+                            # TODO: validate the exception message (how?)
+                            raise Exception(
+                                "The ETag provided for the S3BodyLocation does not match the S3 Object"
+                            )
```

Comment:
It should lead to a `CREATE_FAILED` state and checking the stack events should indicate why the resource failed to deploy. Currently though this is not yet supported very well, so for now I'd say it's ok if this isn't covered specifically via a test :+1: